### ISSUE TITLE
Possibly Document the format of various DataSerializables

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/SpongeDocs#422]
As much as the [MinecraftWiki](http://minecraft.gamepedia.com/Chunk_format) goes into depth elaborating on the format for various objects when serialized to NBT, we should likely do the same so that plugin developers can safely look at the docs and understand what data is available where and when and what content version.

Thoughts?

**Comment by Inscrutable** [comment]
You are suggesting we should reproduce/rewrite the MC Wiki pages on serialization here?
One of the docs guidelines is to minimise duplication, linking to external resources rather than reproducing them. This keeps the size of the docs (and resulting translations) down, and less maintenance is required. We make exceptions to permit translation, where the source has little to offer, or is poorly organised.
So, what is the case for reproducing these NBT references?

**Comment by gabizou** [comment]
I'm suggesting since we have much of our own format for a lot of the `DataSerializable`s in common, there's a lot of "gotchas" in certain cases, but to put it plainly, explaining what can be found in a [`BlockSnapshot`](https://github.com/SpongePowered/SpongeCommon/blob/master/src/main/java/org/spongepowered/common/block/SpongeBlockSnapshot.java#L205)'s `DataContainer` is quite handy for plugin developers.

**Comment by saladoc** [comment]
Yes, it might be useful. However I can't imagine too many people using this, so it will have to wait until some of the heavier stuff is finished.
